adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -102,91 +100,73 @@ - (void)presentFromViewController:(nonnull UIViewController *)viewController {
+  dispatch_async(_lockQueue, ^{
+    GADMAdapterUnityMapTableRemoveObjectForKey(_placementInUseRewarded, self->_placementID);
+  });
 
-    // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
-    // amount of 1. Publishers using this adapter should override the reward on the AdMob
-    // front-end.
-    GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
-                                                     rewardAmount:[NSDecimalNumber one]];
-    [_adEventDelegate didRewardUserWithReward:reward];
-  } else if (state == kUnityAdsFinishStateError) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorFinish,
-        @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
-    [_adEventDelegate didFailToPresentWithError:error];
+  if (_adLoadCompletionHandler) {
+    NSError *error = GADUnityErrorWithDescription([NSString
+        stringWithFormat:@"Failed to load rewarded ad with placement ID '%@'", placementId]);
+    _adEventDelegate = _adLoadCompletionHandler(nil, error);
   }
-
-  [_adEventDelegate willDismissFullScreenView];
-  [_adEventDelegate didDismissFullScreenView];
+  [self cleanup];
 }
 
-- (void)unityAdsDidStart:(nonnull NSString *)placementID {
-  [_adEventDelegate didStartVideo];
-}
+#pragma mark - UnityAdsShowDelegate Methods
 
-- (void)unityAdsReady:(nonnull NSString *)placementID {
-  // Logic to mark a placement ready has moved to the UnityAdsLoadDelegate function
-  // unityAdsAdLoaded.
+- (void)unityAdsShowStart:(nonnull NSString *)placementId {
+  [_adEventDelegate didStartVideo];
 }
 
-- (void)unityAdsDidClick:(nonnull NSString *)placementID {
+- (void)unityAdsShowClick:(NSString *)placementId {
   // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
   // that a click event indicates the user is leaving the application for a browser or deeplink, and
   // notifies the Google Mobile Ads SDK accordingly.
   [_adEventDelegate reportClick];
 }
 
-- (void)unityAdsPlacementStateChanged:(nonnull NSString *)placementID
-                             oldState:(UnityAdsPlacementState)oldState
-                             newState:(UnityAdsPlacementState)newState {
-}
-
-#pragma mark - UnityAdsLoadDelegate Methods
-
-- (void)unityAdsAdFailedToLoad:(nonnull NSString *)placementId {
-  dispatch_async(_lockQueue, ^{
-    GADMAdapterUnityMapTableRemoveObjectForKey(_placementInUseRewarded, self->_placementID);
-  });
+- (void)unityAdsShowComplete:(NSString *)placementId withFinishState:(UnityAdsShowCompletionState)state {
+  [_adEventDelegate didEndVideo];
 
-  if (_adLoadCompletionHandler) {
-    NSError *error = GADUnityErrorWithDescription([NSString
-        stringWithFormat:@"Failed to load rewarded ad with placement ID '%@'", placementId]);
-    _adEventDelegate = _adLoadCompletionHandler(nil, error);
+  if (state == kUnityAdsFinishStateCompleted) {
```

Comment:
Rewarded is been skipped after completing videos.  
State is returning kUnityShowComplationStateCompleted 
<img width="467" alt="Screen Shot 2021-04-01 at 1 28 26 PM" src="https://user-images.githubusercontent.com/59298836/113350358-3260ab80-92ee-11eb-8a45-87b814d3da45.png">
<img width="447" alt="Screen Shot 2021-04-01 at 1 29 23 PM" src="https://user-images.githubusercontent.com/59298836/113350435-4efce380-92ee-11eb-942f-f3c65f95bde4.png">

Reply:
Thanks @chlandacastro, I'll look into this.

Reply:
Thanks again @chlandacastro, should be fixed now.